compiler/rustc_session/src/session.rs

```diff
@@ -907,7 +907,7 @@ impl Session {
                 | CrateType::Rlib
                 | CrateType::Staticlib
                 | CrateType::Cdylib => continue,
-                CrateType::ProcMacro => return false,
+                CrateType::ProcMacro => return true,
```

Comment:
Well I guess this answer the ambiguity in RFC about what to do this proc-macros.

However, I don't think this is the right fix, this would now always remap paths and not take into account `-Zremap-path-scope`, we should instead remove the loop.

```diff
-        // bail out, if any of the requested crate types aren't:
-        // "compiled executables or libraries"
-        for crate_type in &self.opts.crate_types {
-            match crate_type {
-                CrateType::Executable
-                | CrateType::Dylib
-                | CrateType::Rlib
-                | CrateType::Staticlib
-                | CrateType::Cdylib => continue,
-                CrateType::ProcMacro => return false,
-            }
-        }
```

----

Btw, this PR title is a bit misleading, it's about all paths in a proc-macro not just `.rustc`.

Reply:
I changed 'PR' subject, and tried a build by removing the 'loop' and build is success. Will you pull this change / I should do anything else?

Reply:
Great.

> Will you pull this change / I should do anything else?

 Just push a new commit removing the loop.